app/src/lib/stores/app-store.ts

```diff
@@ -3344,49 +3346,87 @@ export class AppStore extends TypedBaseStore<IAppState> {
       const { branchesState } = this.repositoryStateCache.get(r)
-      let branchToCheckout = branchesState.defaultBranch
 
-      // If the default branch is null, use the most recent branch excluding the branch
-      // the branch to delete as the branch to checkout.
-      if (branchToCheckout === null) {
-        let i = 0
-
-        while (i < branchesState.recentBranches.length) {
-          if (branchesState.recentBranches[i].name !== branch.name) {
-            branchToCheckout = branchesState.recentBranches[i]
-            break
-          }
-          i++
-        }
-      }
+      const branchToCheckout = this.getBranchToCheckoutAfterDelete(
+        branch,
+        branchesState
+      )
+      const gitStore = this.gitStoreCache.get(r)
 
-      if (branchToCheckout === null) {
-        throw new Error(
-          `It's not possible to delete the only existing branch in a repository.`
+      if (branchToCheckout != null) {
+        await gitStore.performFailableOperation(() =>
+          checkoutBranch(r, account, branchToCheckout)
         )
       }
 
-      const nonNullBranchToCheckout = branchToCheckout
-      const gitStore = this.gitStoreCache.get(r)
-
       await gitStore.performFailableOperation(() =>
-        checkoutBranch(r, account, nonNullBranchToCheckout)
+        deleteLocalBranch(r, branch, account, includeRemote)
       )
+
+      return this._refreshRepository(r)
+    })
+  }
+
+  /** This shouldn't be called directly. See `Dispatcher`. */
+  public async _deleteRemoteBranch(
+    repository: Repository,
+    branch: Branch
+  ): Promise<void> {
+    return this.withAuthenticatingUser(repository, async (r, account) => {
+      const gitStore = this.gitStoreCache.get(r)
+
       await gitStore.performFailableOperation(() =>
-        deleteBranch(r, branch, account, includeRemote)
+        deleteRemoteBranch(r, branch, account)
       )
 
       return this._refreshRepository(r)
     })
   }
 
+  private getBranchToCheckoutAfterDelete(
+    branchToDelete: Branch,
+    branchesState: IBranchesState
+  ): Branch | null {
+    const tip = branchesState.tip
+    const currentBranch = tip.kind === TipState.Valid ? tip.branch : null
+    // if current branch is not the branch being deleted, no need to switch
+    // branches
+    if (currentBranch != null && branchToDelete.name !== currentBranch.name) {
+      return null
+    }
+
+    let branchToCheckout = branchesState.defaultBranch
+
+    // If the default branch is null, use the most recent branch excluding the
+    // branch the branch to delete as the branch to checkout.
+    if (branchToCheckout === null) {
+      let i = 0
+
+      while (i < branchesState.recentBranches.length) {
+        if (branchesState.recentBranches[i].name !== branchToDelete.name) {
+          branchToCheckout = branchesState.recentBranches[i]
+          break
+        }
+        i++
+      }
+    }
+
+    if (branchToCheckout === null) {
```

Comment:
I know this isn't your doing but am I missing something or is this a very convoluted way of saying

```suggestion
    // If the default branch is null, use the most recent branch excluding the branch
    // the branch to delete as the branch to checkout.
    const branchToCheckout =
      branchesState.defaultBranch ??
      branchesState.recentBranches.find(x => x.name !== branch.name)
    
    if (branchToCheckout === undefined) {
```